scripts/deploy-archive

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+declare wasm=
+declare canister_id=
+declare network=
+
+#########
+# USAGE #
+#########
+
+function title() {
+    echo "Deploy the archive to an II canister"
+}
+
+function usage() {
+    cat << EOF
+
+Usage:
+  $0 --wasm PATH --canister-id CANISTER_ID [--network URL]
+
+Options:
+  --wasm PATH                     Path to the archive wasm file to deploy.
+  --canister-id CANISTER_ID       Canister id of the II canister to deploy the archive to.
+  --network URL                   Optional network parameter. Defaults to local replica.
+EOF
+}
+
+function help() {
+    cat << EOF
+
+Deploys the archive to the specified II canister.
+
+NOTE: This requires dfx, hexdump, sed, a running II canister with the appropriate configuration to accept the supplied archive wasm.
+EOF
+}
+
+# ARGUMENT PARSING
+
+while [[ $# -gt 0 ]]
+do
+    case $1 in
+        --help)
+            title
+            usage
+            help
+            exit 0
+            ;;
+        --wasm)
+            wasm="${2:?missing value for '--wasm'}"
+            shift; # shift past --wasm and value
+            shift;
+            ;;
+        --canister-id)
+            canister_id="${2:?missing value for '--canister-id'}"
+            shift; # shift past --canister-id and value
+            shift;
+            ;;
+        --network)
+            network="${2:?missing value for '--network'}"
+            shift; # shift past --network and value
+            shift;
+            ;;
+        *)
+            echo "ERROR: unknown argument $1"
+            usage
+            echo
+            echo "Use 'release --help' for more information."
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$wasm" ]
+then
+    echo no wasm
+    usage
+    exit 1
+fi
+
+if [ -z "$canister_id" ]
+then
+    echo no canister id
+    usage
+    exit 1
+fi
+
+declare -a network_arg=( )
+
+if [ "$network" ]
+then
+    network_arg+=( "--network" "$network")
+fi
+
+tmp_arg_file=$(mktemp)
+
+echo -n '(blob "' > "$tmp_arg_file"
+hexdump -ve '1/1 "%.2x"' "$wasm" | sed 's/../\\&/g' >> "$tmp_arg_file"
+echo -n '")' >> "$tmp_arg_file"
+
+dfx canister "${network_arg[@]}" call "$canister_id" deploy_archive --argument-file "$tmp_arg_file"
+
+rm "$tmp_arg_file"
```

Comment:
I'm not sure what's happening here, couple of questions:

* Why use `--argument-file` as opposed to `--argument`? I remember discussion about `--argument` being broken sometimes, is that what's happening here? Why does `--argument-file` work differently? Is there a ticket you can share?
* Why the temporary file business?

Since I don't understand what's happening I'm not sure this is relevant, but here's a snippet that I think encodes the argument like you do but without tmp file:

```bash
echo "(blob \"$(hexdump -ve '1/1 "%.2x"' ./archive.wasm | sed 's/../\\&/g')\")"
cat <(echo "(blob \"$(hexdump -ve '1/1 "%.2x"' ./archive.wasm | sed 's/../\\&/g')\")")

```

both give `(blob "\66...")`, once used as argument, once as file argument. And speaking of response files, doesn't `dfx` support `@...` response file syntax? (like gcc, curl, etc)

Reply:
* `--argument-file` is used because the full wasm is quite big and resulted in an error (`argument list too long`) when running the script locally. Using a file solves the issue.
* The scripts creates a temporary file so that people don't get their environment cluttered when running the script.
* AFAIK `dfx` does not support response files